pkg/provider/pods.go

```diff
@@ -379,3 +382,42 @@ func (p *Pod) IsUsingSRIOV() (bool, error) {
 
 	return false, nil
 }
+
+func (p *Pod) IsUsingClusterRoleBinding(clusterRoleBindings []rbacv1.ClusterRoleBinding) (bool, error) {
+	// This function accepts a list of clusterRoleBindings and checks to see if the pod's service account is
+	// tied to any of them.  If it is, then it returns true, otherwise it returns false.
+
+	// Short circuit.  If there is no serviceAcccountName set we can skip the rest of the checks.
+	// Note: This currently does not handle where projected volumes are used to access service account tokens.
+	if p.Pod.Spec.ServiceAccountName == "" {
+		return false, nil
+	}
+
+	// Get all of the service accounts in the pod's namespace.
+	serviceAccounts, err := rbac.GetServiceAccountsFromNamespace(p.Pod.Namespace)
```

Comment:
Is this really necessary? As far as I know, If the pod has a ServiceAccountName, that SA must exist inside the pod's namespace, otherwise the pod wouldn't have been deployed.
```
	podServiceAccountName := p.Pod.Spec.ServiceAccountName

	for crbIndex := range clusterRoleBindings {
		for _, subject := range clusterRoleBindings[crbIndex].Subjects {
			if subject.Kind == rbacv1.ServiceAccountKind && subject.Name == podServiceAccountName && subject.Namespace == p.Namespace{
				tnf.ClaimFilePrintf("Pod %s has service account %s that is tied to cluster role binding %s", p.Pod.Name, podServiceAccountName, clusterRoleBindings[crbIndex].Name)
				return true, nil
			}
		}
	}
```
Also note that I also added the check for the subject.Kind and the namespace, as SA's can have the same name across namespaces.

Reply:
This should be addressed in my latest commit.